src/dsl/static.jl

```diff
@@ -66,184 +66,172 @@ function parse_julia_expr!(stmts, bindings, name::Symbol, var::Symbol,
         # don't create a new Julia node, just use the existing node
         return bindings[var]
     end
-    parse_julia_expr!(stmts, bindings, name, Expr(:block, var), typ)
+    return parse_julia_expr!(stmts, bindings, name, Expr(:block, var), typ)
 end
 
 function parse_julia_expr!(stmts, bindings, name::Symbol, var::QuoteNode,
                            typ::Union{Symbol,Expr,QuoteNode})
     fn = Expr(:function, Expr(:tuple), var)
-    node = gensym()
+    node = gensym(name)
     push!(stmts, :($(esc(node)) = add_julia_node!(
         builder, $fn, inputs=[], name=$(QuoteNode(name)),
         typ=$(QuoteNode(typ)))))
-    node
+    return node
 end
 
 function parse_julia_expr!(stmts, bindings, name::Symbol, value,
                            typ::Union{Symbol,Expr,QuoteNode})
     fn = Expr(:function, Expr(:tuple), QuoteNode(value))
-    node = gensym()
+    node = gensym(name)
     push!(stmts, :($(esc(node)) = add_julia_node!(
         builder, $fn, inputs=[], name=$(QuoteNode(name)),
         typ=$(QuoteNode(typ)))))
-    node
+    return node
 end
 
-function parse_julia_assignment!(stmts, bindings, line::Expr)
-    if line.head != :(=)
-        return false
+function parse_assignment!(stmts, bindings, lhs, rhs)
+    if isa(lhs, Expr) && lhs.head == :tuple
+        # Recursively handle tuple assignments
+        name, typ = gen_node_name(rhs), QuoteNode(Any)
+        node = parse_julia_expr!(stmts, bindings, name, rhs, typ)
+        bindings[name] = node
+        for (i, lhs_i) in enumerate(lhs.args)
+            # Assign lhs[i] = rhs[i]
+            rhs_i = :($name[$i])
+            parse_assignment!(stmts, bindings, lhs_i, rhs_i)
+        end
+    else
+        # Handle single variable assignment (base case)
+        (name::Symbol, typ) = parse_typed_var(lhs)
+        # Generate new node name if name is already bound
+        node_name = haskey(bindings, name) ? gensym(name) : name
+        node = parse_julia_expr!(stmts, bindings, node_name, rhs, typ)
+        # Old bindings are overwritten with new nodes
+        bindings[name] = node
```

Comment:
@ztangent Is this the code that allows you to reassign a variable?

Reply:
Yup! What's happening here is that the first time the variable is assigned, a `JuliaNode` gets created with the same node name as the name of the variable. But when it gets reassigned, a new `JuliaNode` is created with a gensym-ed name, and the binding for that variable gets replaced by the newly created node, since all future references to that variable will refer to that node* .

(*Assuming straight-line code, which is a restriction of the static DSL. If you do something weird like an assignment within a let block, the parser doesn't know that the assignment is within a different scope, so errors will occur.)